src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -0,0 +1,191 @@
+  });
+  const [createPublicKey] = useMutation<
+    CreatePublicKeyMutation,
+    CreatePublicKeyMutationVariables
+  >(CREATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error creating the public key: ${error.message}`
+      );
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.createPublicKey] },
+      });
+    },
+  });
+
+  const [keyName, setKeyName] = useState(initialKeyName ?? "");
+  const [keyValue, setKeyValue] = useState(initialKeyValue ?? "");
+
+  useEffect(() => {
+    setKeyName(initialKeyName);
+  }, [initialKeyName]);
+
+  useEffect(() => {
+    setKeyValue(initialKeyValue);
+  }, [initialKeyValue]);
+
+  useEffect(() => {
+    const inputErrors = [];
+    if (!keyName) {
+      inputErrors.push(EMPTY_KEY_NAME);
+    }
+    if (
+      existingKeys?.find(({ name }) => name === keyName) &&
+      keyName !== replaceKeyName
+    ) {
+      inputErrors.push(DUPLICATE_KEY_NAME);
+    }
+    const hasRSA = keyValue?.substring(0, SSH_RSA.length) === SSH_RSA;
+    const hasDSS = keyValue?.substring(0, SSH_DSS.length) === SSH_DSS;
+    if (!hasRSA && !hasDSS) {
+      inputErrors.push(INVALID_SSH_KEY);
+    }
+    setErrors(inputErrors);
+  }, [keyName, keyValue, replaceKeyName, existingKeys]);
+
+  const onClickSave = () => {
+    const nextKeyInfo = { name: keyName, key: keyValue };
+    if (replaceKeyName) {
+      updatePublicKey({
+        variables: { targetKeyName: replaceKeyName, updateInfo: nextKeyInfo },
+      });
+    } else {
+      createPublicKey({ variables: { publicKeyInput: nextKeyInfo } });
+    }
+    onCancel();
+  };
+
+  return (
+    <Modal
+      data-cy="key-edit-modal"
+      visible={visible}
+      onCancel={onCancel}
+      footer={
+        <>
+          <LeftButton
+            key="cancel"
+            onClick={onCancel}
+            data-cy="cancel-subscription-button"
+          >
+            Cancel
+          </LeftButton>
+          <Button
+            key="save"
+            data-cy="save-key-button"
+            disabled={errors.length > 0}
+            onClick={onClickSave}
+            variant={Variant.Primary}
+          >
+            Save
+          </Button>
+        </>
+      }
+      title={replaceKeyName ? "Update Public Key" : "Add Public Key"}
+    >
+      <InputLabel>Key Name</InputLabel>
+      <StyledInput
+        value={keyName}
+        onChange={(e) => {
+          setKeyName(e.target.value);
+        }}
+        data-cy="key-name-input"
+      />
+      <InputLabel>Public Key</InputLabel>
+      <TextArea
+        data-cy="key-value-input"
+        value={keyValue}
+        autoSize={{ minRows: 4, maxRows: 6 }}
+        onChange={(e) => setKeyValue(e.target.value)}
+      />
+      <ErrorContainer>
+        {errors.map((text) => (
+          <div key={uuid()} data-cy="error-message">
```

Comment:
Why have an unstyled div wrapping the error message?

Reply:
Since ErrorMessage extends the Body tag, I need to wrap them with a div so they go on a new line. I could also use <br/>

Reply:
div is ok